### PR TITLE
fix nil pointer in dameonset controller

### LINKED ISSUE #21933: nil pointer dereference in daemon controller
https://pantheon.corp.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/21929/kubernetes-pull-build-test-e2e-gce/30298/
```
E0224 23:27:08.591768       6 runtime.go:58] Recovered from panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/util/runtime/runtime.go:52
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/util/runtime/runtime.go:40
/usr/src/go/src/runtime/asm_amd64.s:402
/usr/src/go/src/runtime/panic.go:387
/usr/src/go/src/runtime/panic.go:42
/usr/src/go/src/runtime/sigpanic_unix.go:26
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/daemon/controller.go:298
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/daemon/controller.go:161
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/framework/controller.go:178
<autogenerated>:25
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/framework/controller.go:248
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/framework/controller.go:122
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/framework/controller.go:97
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/util/wait/wait.go:66
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/util/wait/wait.go:67
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/util/wait/wait.go:47
```

@mikedanese

### PR DESCRIPTION
**Pull request by mikedanese**
Fixes #21933   [edited by @wojtek-t]

cc @janetkuo @bprashanth

### CODE REVIEW

**Comment by davidopp** [comment]
LGTM

Thanks!

BTW can you update the first comment in the thread? The issue number you listed is actually this PR.

**Comment by wojtek-t** [comment]
Actually it fixes #21933

File: pkg/controller/daemon/controller.go [hunk 1]
```diff
@@ -312,6 +312,7 @@ func (dsc *DaemonSetsController) deletePod(obj interface{}) {
 			return
 		}
 	}
+	glog.V(4).Infof("Pod %s deleted.", pod.Name)
```
Comment (lavalamp) [inline pkg/controller/daemon/controller.go]: This could still be nil if you got a tombstone, right?

Comment (bprashanth) [inline pkg/controller/daemon/controller.go]: That should hit one of the code paths above, and retrieve the tombstone.obj, which if it isn't a pod should return

Comment (mikedanese) [inline pkg/controller/daemon/controller.go]: Yup :) should be set be this point

Comment (mikedanese) [inline pkg/controller/daemon/controller.go]: It's either set on line 309 or the function returns before we hit this.

Comment (lavalamp) [inline pkg/controller/daemon/controller.go]: Oh, sorry, somehow I didn't notice that github omitted the line.

**Pull request merged**